esmvalcore/preprocessor/_shared.py

```diff
@@ -181,3 +182,73 @@ def update_weights_kwargs(
     else:
         kwargs.pop('weights', None)
     return kwargs
+
+
+def get_normalized_cube(
+    cube: Cube,
+    statistics_cube: Cube,
+    normalize_with_stats: Literal['subtract', 'divide'],
+) -> Cube:
+    """Get cube normalized with statistics cube.
+
+    Parameters
+    ----------
+    cube:
+        Input cube that will be normalized.
+    statistics_cube:
+        Cube that is used to normalize the input cube. Needs to be
+        broadcastable to the input cube's shape according to iris' rich
+        broadcasting rules enabled by the use of named dimensions (see also
+        https://scitools-iris.readthedocs.io/en/latest/userguide/cube_maths.
+        html#calculating-a-cube-anomaly). This is usually ensure by using
+        :meth:`iris.cube.Cube.collapsed` to calculate the statistics cube.
+    normalize_with_stats:
+        Normalization operation. Can either be `subtract` (statistics cube is
+        subtracted from the input cube) or `divide` (input cube is divided by
+        the statistics cube).
+
+    Returns
+    -------
+    Cube
+        Input cube normalized with statistics cube.
+
+    """
+    if normalize_with_stats == 'subtract':
+        normalized_cube = cube - statistics_cube
+
+    elif normalize_with_stats == 'divide':
+        # Raise proper zero-division warning if necessary
+        if (statistics_cube.core_data() == 0.0).any():
```

Comment:
This kind of check can be computationally expensive if the cube is large. Instead of checking and warning at runtime, I would recommend just documenting the behaviour in the docstring (i.e. write that normalization with 'divide' will mask invalid values) and not do any checks at runtime because you're going to do this anyway and chances are real that users will not see the warning anyway in between the hundreds of warnings coming from iris.